api/src/main/java/graphql/nadel/normalized/ExecutableNormalizedOperationToAstCompiler.java

```diff
@@ -0,0 +1,189 @@
+package graphql.nadel.normalized;
+
+import graphql.Internal;
+import graphql.com.google.common.collect.ImmutableList;
+import graphql.com.google.common.collect.ImmutableMap;
+import graphql.language.Argument;
+import graphql.language.ArrayValue;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.InlineFragment;
+import graphql.language.NullValue;
+import graphql.language.ObjectField;
+import graphql.language.ObjectValue;
+import graphql.language.OperationDefinition;
+import graphql.language.Selection;
+import graphql.language.SelectionSet;
+import graphql.language.TypeName;
+import graphql.language.Value;
+import graphql.nadel.normalized.ValueToVariableValueCompiler.VariableValueWithDefinition;
+import graphql.normalized.ExecutableNormalizedField;
+import graphql.normalized.NormalizedInputValue;
+import graphql.schema.GraphQLSchema;
+import kotlin.Pair;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static graphql.collect.ImmutableKit.map;
+import static graphql.language.Argument.newArgument;
+import static graphql.language.Field.newField;
+import static graphql.language.InlineFragment.newInlineFragment;
+import static graphql.language.SelectionSet.newSelectionSet;
+import static graphql.language.TypeName.newTypeName;
+import static graphql.nadel.normalized.ValueToVariableValueCompiler.normalizedInputValueToVariable;
+
+@Internal
+public class ExecutableNormalizedOperationToAstCompiler {
+
+    private static final String JSON_SCALAR_TYPENAME = "JSON";
+
+    public static Pair<Document, Map<String, Object>> compileToDocument(
```

Comment:
I'm going to assume the only thing you changed in this class after copying was make the return `Pair` and update `argValue` to generate variables? (+ new variables List)

Reply:
that's correct